Seqan 1.4.1 not compilable with gcc 4.6 to 5.x

```
/home/sachsenb/OpenMS/contrib-build/include/seqan/graph_types/graph_impl_fragment.h: In function 'bool seqan::operator<(const seqan::Fragment<TSize, seqan::ExactReversableFragment<TSpec> >&, const seqan::Fragment<TSize, seqan::ExactReversableFragment<TSpec> >&)':
/home/sachsenb/OpenMS/contrib-build/include/seqan/graph_types/graph_impl_fragment.h:296:14: error: parse error in template argument list
    if (left.reversed < right.reversed)
```
This is a compiler bug. He mistakes the smaller than operator with the start of a template. Fixed in gcc6. This is also present in the aptitude package unfortunately. Seqan 1.4.1 is not maintained anymore but moving to seqan 2 or now even 3 would be a major effort.

Refs:
https://svn.boost.org/trac10/ticket/13386
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=10200

--- a/libraries.cmake/seqan.cmake
+++ b/libraries.cmake/seqan.cmake
@@ -14,6 +14,11 @@ MACRO( OPENMS_CONTRIB_BUILD_SEQAN )
   endif()
   OPENMS_SMARTEXTRACT(ZIP_ARGS ARCHIVE_SEQAN "SEQAN" "include/seqan/version.h")
 
+  # https://github.com/OpenMS/contrib/issues/81
+  set(PATCH_FILE "${PROJECT_SOURCE_DIR}/patches/seqan/graph_impl_fragment_smaller_than_template.diff")
+  set(PATCHED_FILE "${SEQAN_DIR}/graph_types/graph_impl_fragment.h")
+  OPENMS_PATCH( PATCH_FILE SEQAN_DIR PATCHED_FILE)
+
   # create seqan build system and install to target directory
   set(_SEQAN_BUILD_DIR "${SEQAN_DIR}/build")
   file(TO_NATIVE_PATH "${_SEQAN_BUILD_DIR}" _SEQAN_NATIVE_BUILD_DIR)

--- a/libraries.cmake/seqan.cmake
+++ b/libraries.cmake/seqan.cmake
@@ -16,7 +16,7 @@ MACRO( OPENMS_CONTRIB_BUILD_SEQAN )
 
   # https://github.com/OpenMS/contrib/issues/81
   set(PATCH_FILE "${PROJECT_SOURCE_DIR}/patches/seqan/graph_impl_fragment_smaller_than_template.diff")
-  set(PATCHED_FILE "${SEQAN_DIR}/graph_types/graph_impl_fragment.h")
+  set(PATCHED_FILE "${SEQAN_DIR}/core/include/seqan/graph_types/graph_impl_fragment.h")
   OPENMS_PATCH( PATCH_FILE SEQAN_DIR PATCHED_FILE)
 
   # create seqan build system and install to target directory

--- a/libraries.cmake/seqan.cmake
+++ b/libraries.cmake/seqan.cmake
@@ -14,6 +14,7 @@ MACRO( OPENMS_CONTRIB_BUILD_SEQAN )
   endif()
   OPENMS_SMARTEXTRACT(ZIP_ARGS ARCHIVE_SEQAN "SEQAN" "include/seqan/version.h")
 
+  # required for compiler bug in GCC 4-5.X (seems to be fixed with GCC6)
   # https://github.com/OpenMS/contrib/issues/81
   set(PATCH_FILE "${PROJECT_SOURCE_DIR}/patches/seqan/graph_impl_fragment_smaller_than_template.diff")
   set(PATCHED_FILE "${SEQAN_DIR}/core/include/seqan/graph_types/graph_impl_fragment.h")

--- a/patches/seqan/graph_impl_fragment_smaller_than_template.diff
+++ b/patches/seqan/graph_impl_fragment_smaller_than_template.diff
@@ -1,0 +1,11 @@
+--- graph_types/graph_impl_fragment.h	2018-11-13 17:12:45.000000000 +0100
++++ graph_types/graph_impl_fragment.h	2018-11-13 17:12:06.000000000 +0100
+@@ -293,7 +293,7 @@
+         return true;
+     if (left.len > right.len)
+         return false;
+-    if ((left.reversed) < (right.reversed))
++    if (left.reversed < right.reversed)
+         return true;
+     // if (left.reversed > right.reversed)
+     //     return false;

--- a/patches/seqan/graph_impl_fragment_smaller_than_template.diff
+++ b/patches/seqan/graph_impl_fragment_smaller_than_template.diff
@@ -4,8 +4,8 @@
          return true;
      if (left.len > right.len)
          return false;
--    if ((left.reversed) < (right.reversed))
-+    if (left.reversed < right.reversed)
+-    if (left.reversed < right.reversed)
++    if ((left.reversed) < (right.reversed))
          return true;
      // if (left.reversed > right.reversed)
      //     return false;

--- a/patches/seqan/graph_impl_fragment_smaller_than_template.diff
+++ b/patches/seqan/graph_impl_fragment_smaller_than_template.diff
@@ -1,5 +1,5 @@
---- graph_types/graph_impl_fragment.h	2018-11-13 17:12:45.000000000 +0100
-+++ graph_types/graph_impl_fragment.h	2018-11-13 17:12:06.000000000 +0100
+--- core/include/seqan/graph_types/graph_impl_fragment.h	2018-11-13 17:12:45.000000000 +0100
++++ core/include/seqan/graph_types/graph_impl_fragment.h	2018-11-13 17:12:06.000000000 +0100
 @@ -293,7 +293,7 @@
          return true;
      if (left.len > right.len)

